samples/ipc_call_interface/inc/math.h

```diff
@@ -2,7 +2,51 @@
 // Licensed under the MIT license.
 // See LICENSE file in the project root for full license information.
 
+#include "az_ulib_ipc_api.h"
 #include "az_ulib_result.h"
+#include <stdint.h>
 
+/*
+ * interface definition
+ */
+#define MATH_INTERFACE_NAME "math"
+#define MATH_INTERFACE_VERSION 1
+
+typedef az_ulib_ipc_interface_handle math_handle;
+
+#define MATH_INTERFACE_SUM_METHOD (az_ulib_action_index)0
+#define MATH_INTERFACE_SUB_METHOD (az_ulib_action_index)1
+
+/*
+ * Define sum method on math interface.
+ */
+#define MATH_INTERFACE_SUM_METHOD_NAME "sum"
+
+typedef struct sum_model_in_tag {
+  int32_t a;
+  int32_t b;
+} sum_model_in;
+
+typedef int64_t sum_model_out;
+
+/*
+ * Define subtraction method on math interface.
+ */
+#define MATH_INTERFACE_SUB_METHOD_NAME "sub"
+
+typedef struct sub_model_in_tag {
+  int32_t a;
+  int32_t b;
+} sub_model_in;
```

Comment:
I'm guessing it's for demonstration purposes, but why not merge these into one type (`sum_model_in` and `sub_model_in`)? Saves space and makes interacting with the two easier. Same with `model_out`.

Reply:
This file shall be autogenerated from the metadata. the fact that `sum` and `sub` have the same `in` and `out` is just a coincidence. So, to make it standard, each method in the interface expose it's own model_in and model_out structure.

---
In reply to: [529752242](https://github.com/Azure/azure-ulib-c/pull/53#discussion_r529752242) [](ancestors = 529752242)